src/audio/module_adapter/module/waves.c

```diff
@@ -356,8 +356,8 @@ static int waves_effect_init(struct processing_module *mod)
 	waves_codec->o_format = waves_codec->i_format;
 
 	waves_codec->sample_size_in_bytes = sample_bytes;
-	waves_codec->buffer_bytes = mod->period_bytes;
-	waves_codec->buffer_samples = waves_codec->buffer_bytes / src_fmt->channels /
+	waves_codec->buffer_samples = src_fmt->rate * dev->pipeline->period / 1000000;
```

Comment:
maybe you could add a comment to explain that you're preparing a buffer for 1 period worth of data

Reply:
I really don't like this change. It's unclear and feels wrong, so a better explanation should be mandatory.

Reply:
The comment was added. Hopefully, it is more clear.

Reply:
@CyKuei thanks, it looks clear now to me at least.

Reply:
This is not the correct way to add multi line C comments. Please use something like this:

```
/* Prepare a buffer for 1 period worth of data
 * dev->pipeline->period stands for the scheduling period in us */

Reply:
@dbaluta, modified. Thanks for the input.